install_config/persistent_storage/pod_security_context.adoc

```diff
@@ -11,144 +11,406 @@
+High-availability of storage in the infrastructure is left to the underlying
+storage provider.
+
+Accessing persistent storage requires coordination between the cluster (and/or storage)
+administrator and the end-developer. The cluster admin creates persistent volumes (PVs),
+which abstract the underlying physical storage. The developer creates pods and,
+optionally, persistent volume claims (PVCs), which bind to PVs, based on matching
+criteria, such as capacity. For the admin, granting pods access to PVs involves knowing
+the group ids and/or user id assigned to the actual storage, selinux considerations, and,
+finally, ensuring that these ids are allowed in the range of legal ids defined for the
+project and/or the SCC that matches the requirements of the pod.
+
+Group ids, the user id, and selinux values are defined in the `*SecurityContext*` stanza
+in a pod definition. Group ids are global to the pod and apply to all containers defined
+in the pod. User ids can also be global, or specific to each container. Four stanzas
+control access to volumes:
+link:#fsgroup[`fsGroup`],
+link:#supplemental-groups[`supplementalGroups`],
+link:#user-id[`runAsUser`], and
+link:#selinux[`SELinuxOptions`].
+
+[[scc]]
+==== SCCs, Defaults, and Allowed Ranges
+SCCs influence whether or not a pod is given a default user id, fsgroup id, supplemental
+group id, and selinux label; and, whether or not ids supplied in the pod spec (or in the
+image) will be validated against a range of allowable ids. If validation is required and
+fails then the pod will also fail.
+
+SCCs define strategies, such as `RunAsUser`, `supplementalGroups`, and `fsGroup`. These
+strategies are used to help decide if the pod is authorized. Strategy values set to
+_RunAsAny_ are essentially stating that the pod can do what it wants regarding that
+strategy. Authorization is skipped for that strategy and no Openshift default is produced
+based on that strategy. So ids and selinux labels in the resulting container are based on
+*_container_* defaults and not on Openshift policies. Here's a quick summary of _RunAsAny_:
+
+- any id defined in the pod spec (or image) is allowed,
+- absence of an id in the pod spec (and in the image) results in the container assigning an id,
+which is *root* (0) for docker,
+- no selinux labels are defined, so docker will assign a unique (and generally not useful) label.
+
+For these reasons, SCCs with _RunAsAny_ for id related strategies should be protected so
+that ordinary developers do not have access to the SCC.
+
+On the other hand, SCC strategies set to _MustRunAs_ or _MustRunAsRange_ trigger id validation
+(for id related strategies), and cause default values to be supplied by Openshift to the
+container (when those values are not supplied directly in the pod spec or image).
+
+SCCs may define the range of allowed ids (user, groups). If range checking is required
+(e.g., _MustRunAs_) and the allowable range is not defined in the SCC, then the project
```

Comment:
Paul, I've been able to leave the namespace alone and define ranges in the SCC, eg:
fsGroup:
  type: MustRunAs
  ranges:
  - min: 5000
    max: 6000
...
runAsUser:
  type: MustRunAsRange
  uidRangeMin: 99
  uidRangeMax: 65534
...
supplementalGroups:
  type: MustRunAs
  ranges:
  - min: 5000
    max: 6000


----- Original Message -----
> > +*_container_* defaults and not on Openshift policies. Here's a quick
> > summary of _RunAsAny_:
> > +
> > +- any id defined in the pod spec (or image) is allowed,
> > +- absence of an id in the pod spec (and in the image) results in the
> > container assigning an id,
> > +which is *root* (0) for docker,
> > +- no selinux labels are defined, so docker will assign a unique (and
> > generally not useful) label.
> > +
> > +For these reasons, SCCs with _RunAsAny_ for id related strategies should
> > be protected so
> > +that ordinary developers do not have access to the SCC.
> > +
> > +On the other hand, SCC strategies set to _MustRunAs_ or _MustRunAsRange_
> > trigger id validation
> > +(for id related strategies), and cause default values to be supplied by
> > Openshift to the
> > +container (when those values are not supplied directly in the pod spec or
> > image).
> > +
> > +SCCs may define the range of allowed ids (user, groups). If range checking
> > is required
> > +(e.g., _MustRunAs_) and the allowable range is not defined in the SCC,
> > then the project
> 
> The range is defined only on the project, as far as I know -- @pweil-?
> 
> ---
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/openshift-docs/pull/1726/files/9a5c4affe7d76091de47c6b741c73c6702c25c43#r57159436